server/src/resolvers/getVariantsResolver/utils/annotateGnomad.ts

```diff
@@ -1,67 +1,87 @@
+  annotations: (GnomadGRCh37ExomeAnnotation | GnomadGenomeAnnotation)[]
+) => Object.fromEntries(annotations.map(a => [`${a.ref}-${a.pos}-${a.chrom}`, a]));
 
 export const annotate = (
   queryResponse: VariantQueryDataResult[],
   annotations: GnomadAnnotations
 ) => {
-  const { exomeAnnotations, genomeAnnotations } = annotations;
-  const exomeAnnotationKeyMap = _mapToAnnotationsKeyMap(exomeAnnotations);
-  const genomeAnnotationKeyMap = _mapToAnnotationsKeyMap(genomeAnnotations);
+  const { primaryAnnotations, secondaryAnnotations } = annotations;
+  const primaryAnnotationKeyMap = _mapToAnnotationsKeyMap(primaryAnnotations);
+  const secondaryAnnotationKeyMap = _mapToAnnotationsKeyMap(secondaryAnnotations);
 
   queryResponse.forEach(r => {
-    const variantKey = `${r.variant.ref}-${r.variant.start}-${
-      r.variant.chromosome
-    }-${r.variant.assemblyId.replace(/\D/g, '')}`;
+    const variantKey = `${r.variant.ref}-${r.variant.start}-${`${
+      r.variant.assemblyIdCurrent === 'GRCh38' ? 'chr' : ''
+    }${r.variant.chromosome}`}`;
 
-    if (variantKey in exomeAnnotationKeyMap) {
+    if (variantKey in primaryAnnotationKeyMap) {
       /* eslint-disable @typescript-eslint/no-unused-vars */
       const {
-        af: exomeAF,
+        af: primaryAF,
         alt,
-        assembly,
         cdna,
         chrom,
-        ref,
-        pos,
-        type,
         nhomalt,
+        pos,
+        ref,
+        transcript,
         ...rest
-      } = exomeAnnotationKeyMap[variantKey];
+      } = primaryAnnotationKeyMap[variantKey];
       /* eslint-enable @typescript-eslint/no-unused-vars */
-      const genomeAF = genomeAnnotationKeyMap?.[variantKey]?.af ?? 0;
+      // For the GRCh38 assembly, there aren't any secondary gnomAD annotations, so secondaryAF will always be 0
+      // and the overall allele frequency will simply be the genome allele frequency
+      const secondaryAF = secondaryAnnotationKeyMap?.[variantKey]?.af ?? 0;
 
       r.variant.info = {
         ...r.variant.info,
-        // The gnomAD allele frequency is calculated as the highest value between the gnomAD exome allele frequency and the gnomAD genome allele frequency
-        af: Math.max(exomeAF, genomeAF),
+        // The overall allele frequency is calculated as the greater value between the exome allele frequency and the genome allele frequency
+        af: Math.max(primaryAF, secondaryAF),
         gnomadHom: nhomalt,
+        // Ideally, the cdna value should come from the CADD annotations (if available),
+        // but it can also be determined using the values from gnomAD as a fallback
         cdna:
           r.variant.info?.cdna && r.variant.info?.cdna !== 'NA'
             ? r.variant.info?.cdna
             : cdna
             ? `c.${cdna}${ref}>${alt}`
             : 'NA',
+        transcript:
+          r.variant.info?.transcript && r.variant.info?.transcript !== 'NA'
+            ? r.variant.info?.transcript
+            : transcript ?? '',
```

Comment:
According to the Columns Mapping in the[ Gantt Chart ](https://docs.google.com/spreadsheets/d/1eztLBGeqjYj3IafpPxhZo5mbvz6Xzw0LbP_9_HpxeHA/edit#gid=2130371978), `cdna` and `transcript` would come from "CADD". Should we use gnomAD as fallback or remove it to avoid confusion? We haven't verified whether the values for "cdna" and "transcript" make sense. Especially for "transcript", we had extensive conversation regarding which "transcript" value to report with CADD annotations.